src/clp_ffi_js/ir/StreamReader.cpp

```diff
@@ -48,204 +33,40 @@ auto StreamReader::create(DataArrayTsType const& data_array) -> StreamReader {
-        auto const& log_event = m_encoded_log_events[log_event_idx];
-        if (std::ranges::find(
-                    filter_levels,
-                    clp::enum_to_underlying_type(log_event.get_log_level())
-            )
-            != filter_levels.end())
-        {
-            m_filtered_log_event_map->emplace_back(log_event_idx);
-        }
-    }
-}
-
-auto StreamReader::deserialize_stream() -> size_t {
-    if (nullptr == m_stream_reader_data_context) {
-        return m_encoded_log_events.size();
-    }
-
-    constexpr size_t cDefaultNumReservedLogEvents{500'000};
-    m_encoded_log_events.reserve(cDefaultNumReservedLogEvents);
-
-    while (true) {
-        auto result{m_stream_reader_data_context->get_deserializer().deserialize_log_event()};
-        if (result.has_error()) {
-            auto const error{result.error()};
-            if (std::errc::no_message_available == error) {
-                break;
-            }
-            if (std::errc::result_out_of_range == error) {
-                SPDLOG_ERROR("File contains an incomplete IR stream");
-                break;
-            }
-            throw ClpFfiJsException{
-                    clp::ErrorCode::ErrorCode_Corrupt,
-                    __FILENAME__,
-                    __LINE__,
-                    "Failed to deserialize: "s + error.category().name() + ":" + error.message()
-            };
-        }
-        auto const& log_event = result.value();
-        auto const& message = log_event.get_message();
-
-        auto const& logtype = message.get_logtype();
-        constexpr size_t cLogLevelPositionInMessages{1};
-        LogLevel log_level{LogLevel::NONE};
-        if (logtype.length() > cLogLevelPositionInMessages) {
-            // NOLINTNEXTLINE(readability-qualified-auto)
-            auto const log_level_name_it{std::find_if(
-                    cLogLevelNames.begin() + static_cast<size_t>(cValidLogLevelsBeginIdx),
-                    cLogLevelNames.end(),
-                    [&](std::string_view level) {
-                        return logtype.substr(cLogLevelPositionInMessages).starts_with(level);
-                    }
-            )};
-            if (log_level_name_it != cLogLevelNames.end()) {
-                log_level = static_cast<LogLevel>(
-                        std::distance(cLogLevelNames.begin(), log_level_name_it)
-                );
-            }
-        }
+    auto const version{get_version(*zstd_decompressor)};
 
-        auto log_viewer_event{LogEventWithLevel<four_byte_encoded_variable_t>(
-                log_event.get_timestamp(),
-                log_event.get_utc_offset(),
-                message,
-                log_level
+    if (std::ranges::find(cTextIrVersions, version) != cTextIrVersions.end()) {
+        auto stream_reader_data_context{IrStreamReader::create_data_context(
+                std::move(zstd_decompressor),
```

Comment:
I think it's strange for StreamReader to be calling this method:

* The method is `IrStreamReader`-specific since it deals with the `LogEventDeserializer` class which, although it's name doesn't suggest it, is a text-IR-specific class.
* The method is private, so we have to make the base class access a private method of the derived class through the friend access path.

Instead, can we:
* Create `IrStreamReader::create` that handles constructing the `StreamReaderContext` and the `IrStreamReader`.
* Rename `StreamReaderContext` to `IrStreamReaderContext`.
* Replace the logic here with a call to `IrStreamReader::create`.

Reply:
Sorry for missing this earlier. 

> Rename StreamReaderContext to IrStreamReaderContext.
I thought we could templatize class `StreamReaderDataContext` to encapsulate either a text-IR or kvPairs-IR deserializer, which avoids creating a separate data context class for kvPairs-IR. 

I agree with the other two changes.

Reply:
Would you mind renaming it now and then templating it once we add the KV-pair IR stream reader?

Reply:
I see both your points. At the moment, I feel it might not be too bad to leave the name given this class currently resides in path src/clp-ffi-js/ir

Reply:
Sure